accumulator/accumulate_signals_workflow.go

```diff
@@ -0,0 +1,159 @@
+	uniqueGreetingKeysMap := make(map[string]bool)
+	for k, v := range uniqueGreetingKeysMapCAN {
+		uniqueGreetingKeysMap[k] = v
+	}
+	var FirstSignalTime time.Time
+	ExitRequested := false
+
+	ao := workflow.ActivityOptions{
+		StartToCloseTimeout: 10 * time.Second,
+	}
+	ctx = workflow.WithActivityOptions(ctx, ao)
+
+	// Listen to signals in a different goroutine
+	workflow.Go(ctx, func(gCtx workflow.Context) {
+		for {
+			log.Debug("in workflow.go signals goroutine for{} \n")
+			selector := workflow.NewSelector(gCtx)
+			selector.AddReceive(workflow.GetSignalChannel(gCtx, "greeting"), func(c workflow.ReceiveChannel, more bool) {
+				c.Receive(gCtx, &a)
+				unprocessedGreetings = append(unprocessedGreetings, a)
+				log.Debug("Signal Received with text: " + a.GreetingText + ", more: " + strconv.FormatBool(more) + "\n")
+				
+				a = AccumulateGreeting{} 
+			})
+			selector.AddReceive(workflow.GetSignalChannel(gCtx, "exit"), func(c workflow.ReceiveChannel, more bool) {
+				c.Receive(gCtx, nil)
+				ExitRequested = true
+				log.Debug("Exit Signal Received\n")
+			})
+			log.Debug("before select, greeting  text: " + a.GreetingText + "\n")
+			selector.Select(gCtx)
+			log.Debug("after select, greeting  text: " + a.GreetingText + "\n")
+			if FirstSignalTime.IsZero() {
+				FirstSignalTime = workflow.Now(gCtx)
+			}
+		}
+	})
+
+	for ; !workflow.GetInfo(ctx).GetContinueAsNewSuggested() ;{
+		// Wait for Signal or timeout
+		timeout, err := a.GetNextTimeout(ctx, ExitRequested, FirstSignalTime)
+
+		if err != nil {
+			log.Warn("error awaiting signal\n")
+			return "", err
+		}
+		if timeout <= 0 {
+			// do final processing? or just check for signal one more time
+			log.Debug("No time left for signals, checking one more time\n")
+		}
+
+		log.Debug("Awaiting for " + timeout.String() + "\n")
+		gotSignalBeforeTimeout, _ := workflow.AwaitWithTimeout(ctx, timeout, func() bool {
+			return len(unprocessedGreetings) > 0 || ExitRequested
+		})
+
+		// timeout happened without a signal coming in, so let's process the greetings and wrap it up!
+		if len(unprocessedGreetings) == 0 {
+			log.Debug("Into final processing, signal received? " + strconv.FormatBool(gotSignalBeforeTimeout)  + ", exit requested: " + strconv.FormatBool(ExitRequested) +"\n")
+			// do final processing
+			allGreetings = ""
+			err := workflow.ExecuteActivity(ctx, ComposeGreeting, greetings).Get(ctx, &allGreetings)
+			if err != nil {
+				log.Error("ComposeGreeting activity failed.", "Error", err)
+				return allGreetings, err
+			}
+			
+			return allGreetings, nil
+		}
+		
+		/* process latest signal
+		 * Here is where we can process individual signals as they come in.
+		 * It's ok to call activities here.
+		 * This also validates an individual greeting:
+		 * - check for duplicates
+		 * - check for correct bucket 
+		 */
+		for len(unprocessedGreetings) > 0 {
+			ug := unprocessedGreetings[0]
+			unprocessedGreetings = unprocessedGreetings[1:] 
```

Comment:
The way Go works, this is actually a memory leak, but not a big deal because it's in a workflow. Regardless, I would just:

```go
toProcess := unprocessedGreetings
unprocessedGreetings = []AccumulateGreeting{}
for _, greeting := range toProcess {
```

Reply:
nice, done, thank you.